### PR TITLE
community: Fix the bug that Chroma does not specify `embedding_function`

### LINKED ISSUE #18291: Trying to use Chroma vectorstore with default embedding_function results in an error
### Checked other resources

- [x] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

Using this code gives the first type of exception "You must provide an embedding function to compute embeddings."

``` python
import chromadb
from chromadb.utils import embedding_functions
from langchain.chains.query_constructor.base import AttributeInfo
from langchain.retrievers.self_query.base import SelfQueryRetriever
from langchain_openai import ChatOpenAI
from langchain_community.vectorstores import Chroma

metadata_field_info = [
    AttributeInfo(
        name="date",
        description="Date description",
        type="integer",
    ),
]
document_content_description = "Content description"
llm = ChatOpenAI(temperature=0, key)


class ChromaDbInstance:
    def __init__(self) -> None:
        self._client = chromadb.HttpClient(
            host=f"http://localhost:{CHROMA_DB_PORT}",
        )
        self._collection_texts = self._client.get_or_create_collection(
            name=CHROMA_MAIN_COLLECTION_NAME
        )
        self._retriever = SelfQueryRetriever.from_llm(
            llm,
            Chroma(
                client=self._client,
                collection_name=CHROMA_MAIN_COLLECTION_NAME
            ),
            document_content_description,
            metadata_field_info,
        )

    @property
    def count(self):
        return self._collection_texts.count()

    def add_text_to_db(self, text):
        try:
            self._collection_texts.add(
                documents=[text], ids=[str(self.count + 1)]]
            )
        except Exception as e:
            print("-- Error adding new text to db --", e)

    def query_db(self, query):
        return self._retriever.invoke(query)
```

If I pass the "embedding_function" to Chroma initialization - I get another error: "AttributeError: 'ONNXMiniLM_L6_V2' object has no attribute 'embed_query'"

``` python
Chroma(
    client=self._client,
    collection_name=CHROMA_MAIN_COLLECTION_NAME,
    embedding_function=embedding_functions.DefaultEmbeddingFunction(),
),
```

### Error Message and Stack Trace (if applicable)

Initial case when not providing any embedding_function to `langchain_community.vectorstores.Chroma`:
```
Traceback (most recent call last):
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 419, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__
    raise exc
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/routing.py", line 758, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/routing.py", line 778, in app
    await route.handle(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/routing.py", line 299, in handle
    await self.app(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/routing.py", line 79, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/routing.py", line 74, in app
    response = await func(request)
               ^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/fastapi/routing.py", line 278, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/server.py", line 20, in echo_endpoint
    return {"response": db.query_db(req.prompt)}
                        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/modules/vector_db.py", line 72, in query_db
    return self._retriever.invoke(query)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_core/retrievers.py", line 141, in invoke
    return self.get_relevant_documents(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_core/retrievers.py", line 244, in get_relevant_documents
    raise e
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_core/retrievers.py", line 237, in get_relevant_documents
    result = self._get_relevant_documents(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain/retrievers/self_query/base.py", line 186, in _get_relevant_documents
    docs = self._get_docs_with_query(new_query, search_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain/retrievers/self_query/base.py", line 160, in _get_docs_with_query
    docs = self.vectorstore.search(query, self.search_type, **search_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_core/vectorstores.py", line 159, in search
    return self.similarity_search(query, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_community/vectorstores/chroma.py", line 348, in similarity_search
    docs_and_scores = self.similarity_search_with_score(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_community/vectorstores/chroma.py", line 429, in similarity_search_with_score
    results = self.__query_collection(
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_core/utils/utils.py", line 35, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_community/vectorstores/chroma.py", line 155, in __query_collection
    return self._collection.query(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/chromadb/api/models/Collection.py", line 327, in query
    valid_query_embeddings = self._embed(input=valid_query_texts)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/chromadb/api/models/Collection.py", line 629, in _embed
    raise ValueError(
ValueError: You must provide an embedding function to compute embeddings.https://docs.trychroma.com/embeddings
```

Case when providing the default embedding_function to the Chroma initializer
```
Traceback (most recent call last):
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 419, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__
    raise exc
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/routing.py", line 758, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/routing.py", line 778, in app
    await route.handle(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/routing.py", line 299, in handle
    await self.app(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/routing.py", line 79, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/starlette/routing.py", line 74, in app
    response = await func(request)
               ^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/fastapi/routing.py", line 278, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/server.py", line 20, in echo_endpoint
    return {"response": db.query_db(req.prompt)}
                        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/modules/vector_db.py", line 75, in query_db
    return self._retriever.invoke(query)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_core/retrievers.py", line 141, in invoke
    return self.get_relevant_documents(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_core/retrievers.py", line 244, in get_relevant_documents
    raise e
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_core/retrievers.py", line 237, in get_relevant_documents
    result = self._get_relevant_documents(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain/retrievers/self_query/base.py", line 186, in _get_relevant_documents
    docs = self._get_docs_with_query(new_query, search_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain/retrievers/self_query/base.py", line 160, in _get_docs_with_query
    docs = self.vectorstore.search(query, self.search_type, **search_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_core/vectorstores.py", line 159, in search
    return self.similarity_search(query, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_community/vectorstores/chroma.py", line 348, in similarity_search
    docs_and_scores = self.similarity_search_with_score(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/heithvald/Documents/development/projects/project21/python/.venv/lib/python3.11/site-packages/langchain_community/vectorstores/chroma.py", line 437, in similarity_search_with_score
    query_embedding = self._embedding_function.embed_query(query)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ONNXMiniLM_L6_V2' object has no attribute 'embed_query'
```

### Description

* I'm trying to use a SelfQueryRetriever with Chroma vector store. 
* I expect it to work without passing the `embedding_function` arg, or when I pass it explicitly `embedding_function=embedding_functions.DefaultEmbeddingFunction()` to the Chroma constructor
* Instead I get errors when trying to call `retriever.invoke(text)`

I've debugged and found out the problem is most likely in this line: https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/vectorstores/chroma.py#L128 line
If nothing was passed to the `embedding_function` - it would initialize normally and just query the chroma collection and inside the collection it will use the right methods for the embedding_function inside the chromadb lib source code: `return self._embedding_function(input=input)`. At least it will work for the default embedding_function provided by chromadb. Please, fix it.

### System Info

System Information
------------------
> OS:  Darwin
> OS Version:  Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000
> Python Version:  3.11.8 (v3.11.8:db85d51d3e, Feb  6 2024, 18:02:37) [Clang 13.0.0 (clang-1300.0.29.30)]

Package Information
-------------------
> langchain_core: 0.1.27
> langchain: 0.1.9
> langchain_community: 0.0.24
> langsmith: 0.1.10
> langchain_openai: 0.0.8
> chromadb: 0.4.24

### PR DESCRIPTION
**Pull request by liugddx**
- **Issue:** close #18291
- @baskaryan, @eyurtsev PTAL

### CODE REVIEW

File: libs/community/langchain_community/vectorstores/chroma.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
     import chromadb
     import chromadb.config
     from chromadb.api.types import ID, OneOrMany, Where, WhereDocument
+    from chromadb.utils import embedding_functions
```
[the remark this answers is not in the document]
Comment (liugddx) [reply libs/community/langchain_community/vectorstores/chroma.py]: > should this be imported in local scope? this looks like it's only for type checking?

Thanks, I've fixed it

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
Will merge once on my computer

**Comment by baskaryan** [comment]
cc @jeffchuber

**Comment by liugddx** [comment]
It runs fine on my mac

**Comment by sagarmainkar** [comment]
I am still getting this error
Exception has occurred: TypeError langchain_chroma.vectorstores.Chroma() got multiple values for keyword argument 'embedding_function' File "/workspace/src/llama/01_qna.py", line 41, in <module> vectorstore = Chroma.from_documents(texts,embedding_function=[default_ef]) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ TypeError: langchain_chroma.vectorstores.Chroma() got multiple values for keyword argument 'embedding_function'


`import sys
import os

__import__('pysqlite3')
sys.modules['sqlite3'] = sys.modules.pop('pysqlite3')


from langchain_community.document_loaders import DirectoryLoader
from langchain.text_splitter import CharacterTextSplitter
from langchain.embeddings import OpenAIEmbeddings
from langchain_chroma.vectorstores import Chroma
from langchain.chains import create_retrieval_chain
from langchain.chains.combine_documents import create_stuff_documents_chain
from langchain_openai import ChatOpenAI
from langchain import hub
from dotenv import load_dotenv
from chromadb.utils import embedding_functions





load_dotenv()

# sys.path.append(os.path.join(os.getcwd(), '..'))

llm_model = ChatOpenAI(model="gpt-4o-mini", temperature=0)


# loader = TextLoader('news/summary.txt')
loader = DirectoryLoader('news', glob="**/*.txt")

documents = loader.load()
print(len(documents))

text_splitter = CharacterTextSplitter(chunk_size=2500, chunk_overlap=0)
texts = text_splitter.split_documents(documents)

print(texts)
default_ef = embedding_functions.DefaultEmbeddingFunction()
vectorstore = Chroma.from_documents(texts,embedding_function=default_ef)


# See full prompt at https://smith.langchain.com/hub/langchain-ai/retrieval-qa-chat
retrieval_qa_chat_prompt = hub.pull("langchain-ai/retrieval-qa-chat")

combine_docs_chain = create_stuff_documents_chain(
    llm_model, retrieval_qa_chat_prompt)
rag_chain = create_retrieval_chain(
    vectorstore.as_retriever(), combine_docs_chain)


def query(q):
    print("Query: ", q)
    print("Answer: ", rag_chain.invoke({"input": q}))
    print("="*100)


query("What are the effects of legislations surrounding emissions on the Australia coal market?")
query("What are China's plans with renewable energy?")
query("Is there an export ban on Coal in Indonesia? Why?")
query("Who are the main exporters of Coal to China? What is the role of Indonesia in this?")
`

**Pull request merged**